apps/mobile/src/hooks/useMentionableMessage.ts

```diff
@@ -0,0 +1,178 @@
+import { useCallback, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import isFeatureEnabled, { FeatureFlag } from 'src/utils/isFeatureEnabled';
+
+import { useMentionableMessageQueryFragment$key } from '~/generated/useMentionableMessageQueryFragment.graphql';
+import useDebounce from '~/shared/hooks/useDebounce';
+
+type MentionDataType = {
+  interval: {
+    start: number;
+    length: number;
+  };
+  userId?: string;
+  communityId?: string;
+};
+
+export type MentionType = {
+  type: 'User' | 'Community';
+  label: string;
+  value: string;
+};
+
+type Mention = {
+  interval: {
+    start: number;
+    length: number;
+  };
+  userId?: string;
+  communityId?: string;
+};
+
+export function useMentionableMessage(queryRef: useMentionableMessageQueryFragment$key) {
+  const query = useFragment(
+    graphql`
+      fragment useMentionableMessageQueryFragment on Query {
+        ...isFeatureEnabledFragment
+      }
+    `,
+    queryRef
+  );
+
+  const [message, setMessage] = useState('');
+  const [mentions, setMentions] = useState<MentionDataType[]>([]);
+
+  const [isSelectingMentions, setIsSelectingMentions] = useState(false);
+
+  const [aliasKeyword, setAliasKeyword] = useState('');
+  const debouncedAliasKeyword = useDebounce(aliasKeyword, 100);
+
+  const [selection, setSelection] = useState({ start: 0, end: 0 });
+
+  const isMentionEnabled = isFeatureEnabled(FeatureFlag.MENTIONS, query);
+
+  const handleSetMention = useCallback(
+    (mention: MentionType) => {
+      // Use substring to only look up to the current cursor position
+      const upToCursor = message.substring(0, selection.start);
+      const mentionStartPos = upToCursor.lastIndexOf(aliasKeyword);
+      const mentionEndPos = mentionStartPos + aliasKeyword.length;
+
+      const newMessage = `${message.substring(0, mentionStartPos)}@${
+        mention.label
+      } ${message.substring(mentionEndPos)}`;
+
+      setMessage(newMessage);
+
+      const newMention: Mention = {
+        interval: {
+          start: mentionStartPos,
+          length: mention.label.length + 1, // +1 for the @
+        },
+      };
+
+      if (mention.type === 'User') {
+        newMention.userId = mention.value;
+      } else {
+        newMention.communityId = mention.value;
+      }
+
+      // Calculate the length difference between the old alias and the new mention
+      const lengthDifference = mention.label.length + 2 - aliasKeyword.length; // +1 for the @
```

Comment:
+1 or +2 in your comment?

Reply:
Updated the comment! 

It should be + 2